cpp/src/lists/copying/scatter_helper.cu

```diff
@@ -0,0 +1,501 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/detail/get_value.cuh>
+#include <cudf/detail/valid_if.cuh>
+#include <cudf/lists/detail/copying.hpp>
+#include <cudf/lists/detail/scatter_helper.cuh>
```

Comment:
Why do you put the header in `lists/detail` but the cu file in `lists/copying`?

Reply:
It doesn't seem like there's a separation between detail/non-detail files in `src` files.

Reply:
In fact unless the header is needed by more than one module, it can go in the src directory rather than in lists/detail.